### PR TITLE
Enable sygma bridge in khala&phala

### PR DESCRIPTION
**Pull request by tolak**
**Note this is a draft PR, we are still waiting for some runtime config parameters dedicated to Khala&Phala to be confirmed**

This PR is going to enable the Sygma bridge on the Khala&Phala network, some notable configs are:

- PHA Resource ID on Sygma bridge
```sh
pub PHASygmaResourceId: [u8; 32] = hex_literal::hex!("0000000000000000000000000000000000000000000000000000000000000001");
```

- Bridge reserve account(hold bridge liquidity) on Khala&Phala, which is derived by `PalletId`

```sh
pub const SygmaBridgePalletId: PalletId = PalletId(*b"sygma/01");
// SygmaBridgeAccount address: 5EYCAe5jLbHcAAMKvLFSXgCTbPrLgBJusvPwfKcaKzuf5X5e
pub SygmaBridgeAccount: AccountId = SygmaBridgePalletId::get().into_account_truncating();
```

- Bridge fee collection account on Khala be configured through the following line:
```sh
// SygmaBridgeFeeAccountKey Address: 44bdQyeqk5oJzxbZH9xMcovmj3oAxqzSjKujaVhHaZxZuTBH
pub SygmaBridgeFeeAccountKey: [u8; 32] = hex::decode("b00e3e4afb5a9c54036ec6c1775881031fb26b72427a10724c4d8b91099ee889").unwrap().try_into().unwrap();
pub SygmaBridgeFeeAccount: AccountId = SygmaBridgeFeeAccountKey::get().into();
// SygmaBridgeAdminAccountKey Address: 44bdQyeqk5oJzxbZH9xMcovmj3oAxqzSjKujaVhHaZxZuTBH
pub SygmaBridgeAdminAccountKey: [u8; 32] = hex::decode("b00e3e4afb5a9c54036ec6c1775881031fb26b72427a10724c4d8b91099ee889").unwrap().try_into().unwrap();
pub SygmaBridgeAdminAccount: AccountId = SygmaBridgeAdminAccountKey::get().into();
```

- Bridge fee collection account on Phala be configured through the following line:
```sh
// SygmaBridgeFeeAccountKey Address: 44NmbpHjqbz9FcXfVzFUbMFJh5q7qsKAcSTJvFAdYPqQ62Qv
pub SygmaBridgeFeeAccountKey: [u8; 32] = hex::decode("a63f9ccf857e1ab9e806366e3c46ae650de853503d772a987197ab7e22c8f88c").unwrap().try_into().unwrap();
pub SygmaBridgeFeeAccount: AccountId = SygmaBridgeFeeAccountKey::get().into();
// SygmaBridgeAdminAccountKey Address: 44NmbpHjqbz9FcXfVzFUbMFJh5q7qsKAcSTJvFAdYPqQ62Qv
pub SygmaBridgeAdminAccountKey: [u8; 32] = hex::decode("a63f9ccf857e1ab9e806366e3c46ae650de853503d772a987197ab7e22c8f88c").unwrap().try_into().unwrap();
pub SygmaBridgeAdminAccount: AccountId = SygmaBridgeAdminAccountKey::get().into();
```

### CODE REVIEW

**Comment by tolak** [comment]
@mpetrun5 @freddyli7 please help review this PR

**Review by h4x3rotab — APPROVED** [review]
lgtm

File: runtime/phala/src/lib.rs [hunk 1]
```diff
@@ -1547,9 +1557,108 @@ impl chainbridge::Config for Runtime {
...
+    // SygmaBridgeAccount address: 5EYCAe5jLbHcAAMKvLFSXgCTbPrLgBJusvPwfKcaKzuf5X5e
+    pub SygmaBridgeAccount: AccountId = SygmaBridgePalletId::get().into_account_truncating();
+    // SygmaBridgeFeeAccountKey Address: 44NmbpHjqbz9FcXfVzFUbMFJh5q7qsKAcSTJvFAdYPqQ62Qv
+    pub SygmaBridgeFeeAccountKey: [u8; 32] = hex::decode("a63f9ccf857e1ab9e806366e3c46ae650de853503d772a987197ab7e22c8f88c").unwrap().try_into().unwrap();
+    pub SygmaBridgeFeeAccount: AccountId = SygmaBridgeFeeAccountKey::get().into();
+    // SygmaBridgeAdminAccountKey Address: 44NmbpHjqbz9FcXfVzFUbMFJh5q7qsKAcSTJvFAdYPqQ62Qv
+    pub SygmaBridgeAdminAccountKey: [u8; 32] = hex::decode("a63f9ccf857e1ab9e806366e3c46ae650de853503d772a987197ab7e22c8f88c").unwrap().try_into().unwrap();
+    pub SygmaBridgeAdminAccount: AccountId = SygmaBridgeAdminAccountKey::get().into();
+    // EIP712ChainID is the chainID that pallet is assigned with, used in EIP712 typed data domain
+    pub EIP712ChainID: SygmaChainID = U256::from(5231);
```
Comment (freddyli7) [inline runtime/phala/src/lib.rs:1592]: and for Phala, this is `5233`

File: runtime/khala/src/lib.rs [hunk 2]
```diff
@@ -1589,9 +1599,108 @@ impl chainbridge::Config for Runtime {
...
+    // SygmaBridgeAccount address: 5EYCAe5jLbHcAAMKvLFSXgCTbPrLgBJusvPwfKcaKzuf5X5e
+    pub SygmaBridgeAccount: AccountId = SygmaBridgePalletId::get().into_account_truncating();
+    // SygmaBridgeFeeAccountKey Address: 44bdQyeqk5oJzxbZH9xMcovmj3oAxqzSjKujaVhHaZxZuTBH
+    pub SygmaBridgeFeeAccountKey: [u8; 32] = hex::decode("b00e3e4afb5a9c54036ec6c1775881031fb26b72427a10724c4d8b91099ee889").unwrap().try_into().unwrap();
+    pub SygmaBridgeFeeAccount: AccountId = SygmaBridgeFeeAccountKey::get().into();
+    // SygmaBridgeAdminAccountKey Address: 44bdQyeqk5oJzxbZH9xMcovmj3oAxqzSjKujaVhHaZxZuTBH
+    pub SygmaBridgeAdminAccountKey: [u8; 32] = hex::decode("b00e3e4afb5a9c54036ec6c1775881031fb26b72427a10724c4d8b91099ee889").unwrap().try_into().unwrap();
+    pub SygmaBridgeAdminAccount: AccountId = SygmaBridgeAdminAccountKey::get().into();
+    // EIP712ChainID is the chainID that pallet is assigned with, used in EIP712 typed data domain
+    pub EIP712ChainID: SygmaChainID = U256::from(5231);
```
[the remark this answers is not in the document]
Comment (tolak) [reply runtime/khala/src/lib.rs]: Fixed

File: runtime/phala/src/lib.rs (hunk 1 above)
Comment (tolak) [reply runtime/phala/src/lib.rs]: Fixed

**Review by freddyli7 — APPROVED** [review]

**Pull request merged**